x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -211,6 +214,49 @@ export const signalRulesAlertType = ({
           await ruleStatusService.error(errorString);
           wroteStatus = true;
         }
+        const timestampFieldCaps = await services.scopedClusterClient.fieldCaps({
+          index,
+          fields: timestampFields,
+          allow_no_indices: false,
+          include_unmapped: true,
+        });
+
+        if (
```

Comment:
The complexity of this file is very high (above the complexity linter is turned off dangerously so). I would still recommend within this file and any controller based files such as REST based inputs we try to reduce the complexity by pulling functions and code out.

It doesn't have to be with this PR, but the `wroteStatus = true` and the amount of `let` variables is adding up and that is making it more and more difficult for us to maintain this code as well as write unit tests.

If these chunks of code could be moved into more functions to send in inputs and get outputs that help reduce some of the issue we have.

Reply:
As an aside, I think we'll have to move to moving lots of this code into separate functions/chunks of logic to be able to call into for future rule preview work.

Maybe we should bite the bullet on refactoring this file.

Reply:
Yeah I absolutely agree. I think with this PR not getting into BC3 I have a bit of time now to play around with cleaning this up which would benefit everyone long term. I'll take a look at changing this 👍